src/components/common/Modal/index.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect } from 'react';
+import { View } from 'react-native';
+import KeyboardAreaView from 'components/common/KeyboardAreaView';
+import ReactNativeModal from 'react-native-modal';
+import { LOADING, SUCCESS } from '@rootstrap/redux-tools';
+import { bool, func, element, string } from 'prop-types';
+import styles from './styles';
+
+const Modal = ({
+  children,
+  isModalVisible,
+  setModalVisible,
+  status,
+  onBackdropPress = () => {},
+}) => {
+  useEffect(() => {
+    if (status === SUCCESS) {
+      setModalVisible(false);
+    }
+  }, [status]);
+
+  const isLoading = status ? status === LOADING : false;
+
+  return (
+    <ReactNativeModal
+      isVisible={isModalVisible}
+      style={styles.modal}
+      backdropOpacity={0}
+      onBackdropPress={
+        !isLoading
+          ? () => {
+              setModalVisible(false);
+              onBackdropPress();
+            }
+          : () => null
```

Comment:
I think it's better if you define the backdrop function and inside that you use the condition isLoading